client/src/sections/Fridge/Header/Navbar/Navbar.tsx

```diff
@@ -3,19 +3,33 @@ import {useHistory} from 'react-router-dom';
 import {Button, ButtonGroup} from '@shopify/polaris';
 
 import './Navbar.css';
+import {useDispatch} from "react-redux";
+import {logout} from "../../../../store/users/action";
+import {initialState} from "../../../../store/users/reducers";
 
 export default function Navbar() {
   const history = useHistory();
+  const dispatch = useDispatch();
 
+  const redirection = () => {
+    // console.log("Inside Redirection");
+    if(!initialState.isLoggedIn) {
+      alert("You Are Not Logged In!!!");
+      history.push('./login');
+    }
+    return null;
+  };
   const handleHomeClick = () => {
     history.push('./home');
   };
 
   const handleLogoutClick = () => {
     history.push('./login');
+    dispatch(logout());
```

Comment:
You should dispatch the action first before pushing the history. In fact, if the `isLoggedIn` is checked in the `<Fridge />` component you probably don't even need to push to history.

Reply:
The way that I thought about is that when somebody presses logout then shouldn't we logout. I will dispatch before pushing to history.